megahack_installer.sh

```diff
@@ -55,79 +55,73 @@ fi
 
 # check for required packages
 missing_packages=false
-if ! hash unzip 2>/dev/null; then error "unzip is not installed!"; missing_packages=true; fi
+[ ! -x "$(command -v wget)" ] && error "wget is not installed!" && missing_packages=true # some distros don't ship wget by default
+[ ! -x "$(command -v unzip)" ] && error "unzip is not installed!" && missing_packages=true
 if [ "$XDG_SESSION_TYPE" != "wayland" ]; then
-	if ! hash xclip 2>/dev/null; then warn "xclip is not installed, you will have to manually copy the MegaHack path!"; fi
+    [ ! -x "$(command -v xclip)" ] && warn "xclip is not installed, you will have to manually copy the MegaHack path!"
 else
-	if ! hash wl-copy 2>/dev/null; then warn "wl-clipboard is not installed, you will have to manually copy the MegaHack path!"; fi
+    [ ! -x "$(command -v wl-copy)" ] && warn "wl-clipboard is not installed, you will have to manually copy the MegaHack path!"
 fi
 
-echo
-if [ $missing_packages == true ] ; then
+printf "\n"
+if [ $missing_packages == true ]; then
     error "You are missing some programs."
     error "Please install them using your distro's package manager to continue."
     error "Additional information can be found above."
     exit 0
 fi
 
 # if fzf is installed, use it
-if hash fzf 2>/dev/null; then
+if [ -x "$(command -v fzf)" ]; then
     pushd ~ || cd_fail # we want fzf to use the home directory
 
-    if hash fd 2>/dev/null; then
-        export FZF_DEFAULT_COMMAND="fd --extension=zip"
-    else
-        export FZF_DEFAULT_COMMAND="find . -path '*/.*' -prune -o -iname '*.zip' -print"
-    fi
+    [ "$(command -v fd)" ] && export FZF_DEFAULT_COMMAND="fd --extension=zip" || export FZF_DEFAULT_COMMAND="find . -path '*/.*' -prune -o -iname '*.zip' -print"
```

Comment:
this isn't equal to the previous code
shellcheck explains it pretty well: [SC2015](https://www.shellcheck.net/wiki/SC2015)
(this shouldn't cause any issues, but I'm aiming to make this script shellcheck-compliant)